wgengine/tsdns/tsdns.go

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package tsdns provides a Resolver struct capable of resolving
+// domains on a Tailscale network.
+package tsdns
+
+import (
+	"encoding/binary"
+	"errors"
+	"strings"
+	"sync"
+
+	dns "golang.org/x/net/dns/dnsmessage"
+	"inet.af/netaddr"
+	"tailscale.com/types/logger"
+	"tailscale.com/wgengine/packet"
+)
+
+// defaultTTL is the TTL in seconds of all responses from Resolver.
+const defaultTTL = 600
```

Comment:
could just make it 600 * time.Second too and get it back into an `int` later when  needed?